src/main/java/io/pakland/mdas/githubstats/infrastructure/shell/controller/ShellController.java

```diff
@@ -0,0 +1,62 @@
+package io.pakland.mdas.githubstats.infrastructure.shell.controller;
+
+import io.pakland.mdas.githubstats.domain.EntityType;
+import io.pakland.mdas.githubstats.infrastructure.controller.MainController;
+import io.pakland.mdas.githubstats.infrastructure.shell.model.ShellRequest;
+import io.pakland.mdas.githubstats.infrastructure.shell.validation.EntityTypeValidator;
+import io.pakland.mdas.githubstats.infrastructure.shell.validation.YearMonthValidator;
+import io.pakland.mdas.githubstats.infrastructure.shell.validation.NameValidator;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellOption;
+
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+
+@ShellComponent
+public class ShellController {
+
+    private final MainController mainController;
+
+    public ShellController(MainController mainController) {
+        this.mainController = mainController;
+    }
+
+    private boolean build(
+            @ShellOption(value = "entity") String entity,
+            @ShellOption(value = "name") String name,
+            @ShellOption(value = "key") String apiKey,
+            @ShellOption(value = "from") String from,
+            @ShellOption(value = "to") String to,
+            @ShellOption(value = "path", defaultValue = "") String path) {
```

Comment:
Again, this has been simplified in #193 using the Factory Method pattern, which we believe adds more value and maintainability to this controller thing.